pkg/virtctl/root.go

```diff
@@ -55,10 +59,103 @@ func NewVirtctlCommand() *cobra.Command {
 	return rootCmd
 }
 
-func Execute() {
+func Execute(args []string) error {
 	log.InitializeLogging("virtctl")
-	if err := NewVirtctlCommand().Execute(); err != nil {
-		fmt.Println(strings.TrimSpace(err.Error()))
-		os.Exit(1)
+	// check whether there is a MIME config file
+	// if yes, then set up the args for commands and execute
+	// otherwise normal operation
+	confArgs, err := parseMime(args)
+	if err != nil {
+		return err
 	}
+
+	cmd := NewVirtctlCommand()
+	cmd.SetArgs(confArgs)
+
+	if err := cmd.Execute(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func parseMime(args []string) ([]string, error) {
+	var parsedArgs []string
+	var err error
+
+	for ind := 0; ind < len(args); ind++ {
```

Comment:
Ok, this function is very weird now. Let me summarize the  expectations again:
 - If a single argument is supplied which ends in `.vvv`, then it either has to be an existing and valid mime-type file, or we have an error.
 - If it is a single argument which does not end with `.vvv`, then we want a normal execution

Here the code:

```go
func parseMime(args []string) ([]string, error) {
  if len(args) == 2 && strings.HasSuffix(args[1], ".vvv") {
    // retuirns (nil, err) in case of no or an invalid file, (args, nil) otherwise
    return readAndParseMime(args[1])
  }
  return args, nil
}
```

The expectation right now is, that people are logged in to their cluster.

Reply:
The function is weird, because there are optional arguments `--kubeconfig /pat/to/.kubeconfig` and `--server server` which are required in tests.
These two are also required when connecting to the vnc or console.

Thus I have to walk through the parameters and check one by one.

Regarding your expectations, the functions does exactly what you want. It is verified in the tests.

Reply:
Ah I guess you do that because of the functional tests where we pass kubeconfig. You can change them to pass the config parameters via environment variables.

Reply:
Also ok for me, if we assume that the first argument is the mime-file and we simply add any additional flags to the ones in the mime file.

Reply:
I introduced one more check to verify only --kubeconfig and --server gets copied. Otherwise, I made sure original functionality is unharmed.

There is also one think to keep in mind. This patch will get deprecated once next version is released. The current redirection through api server will be insufficient and the proper connection proxy would have to be introduced.

Would you be ok with this implementation for now?